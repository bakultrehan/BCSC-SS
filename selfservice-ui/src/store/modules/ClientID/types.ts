import { PackageAndTest } from '@/models/PackageAndTest';
export interface ClientIdState {
  apiData: any; // PackageAndTest[];
  loading?: boolean;
  successStatus?: boolean;
  errorStatus?: boolean;
  message: string;
}
