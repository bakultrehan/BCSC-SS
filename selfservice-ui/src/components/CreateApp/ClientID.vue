/** * ClientID component */

<template>
  <v-card class="mx-auto mb-5" v-if="getApiData.oidcConfig && getApiData.oidcConfig.clientSecret">
    <v-snackbar v-model="snackbar" :timeout="timeout">
      {{ text }}
      <v-btn color="blue" text @click="snackbar = false">{{ $t('ClientID.labelSnackbarClose') }}</v-btn>
    </v-snackbar>
    <Loading v-if="isLoading && getApiData === 0" />
    <v-toolbar dense class="bc-subtitle-gold-1" dark>
      <v-card-title>{{ $t('ClientID.pagetitle') }}</v-card-title>
    </v-toolbar>
    <v-list dense class="px-5">
      <v-list-item>
        <v-list-item-content>
          {{
          $t('ClientID.titleApiKey')
          }}
          <span
            class="small-hint pad-50"
            v-html="$t('ClientID.titleApiKeyInfo')"
          ></span>
        </v-list-item-content>
        <v-list-item-content
          class="align-end pointer font-weight-bold client-id-copy"
          @click="
                  docopy(
                    getApiData.oidcConfig && getApiData.oidcConfig.clientId
                  )
                "
        >
          <div>
            <v-icon small class="mr-1">mdi-content-copy</v-icon>
            {{ getApiData.oidcConfig && getApiData.oidcConfig.clientId }}
          </div>
        </v-list-item-content>
      </v-list-item>
      <v-divider></v-divider>

      <v-list-item>
        <v-list-item-content>
          {{
          $t('ClientID.titleClientSecret')
          }}
          <span
            class="small-hint pad-50"
            v-html="$t('ClientID.titleClientSecretInfo')"
          ></span>
        </v-list-item-content>
        <v-list-item-content
          class="align-end pointer font-weight-bold d-block"
          @click="
                  docopy(
                    getApiData.oidcConfig && getApiData.oidcConfig.clientSecret
                  )
                "
        >
          <div class="float-left">
            <v-icon small class="mr-2">mdi-content-copy</v-icon>
          </div>
          <div>
            <span>{{ getApiData.oidcConfig && getApiData.oidcConfig.clientSecret }}</span>
          </div>
        </v-list-item-content>
      </v-list-item>

      <v-divider></v-divider>
      <v-list-item>
        <v-list-item-content class="align-self-start">
          {{
          $t('ClientID.titleTestAccount')
          }}
          <span
            class="small-hint pad-50"
            v-html="$t('ClientID.titleTestAccountInfo')"
          ></span>
        </v-list-item-content>
        <v-list-item-content class="align-self-start">
          <v-simple-table dense>
            <template v-slot:default>
              <thead>
                <tr>
                  <th
                    :scope="$t('ClientID.tblCardNo')"
                    class="text-left"
                  >{{$t('ClientID.tblCardNo')}}</th>
                  <th
                    :scope="$t('ClientID.tblPassCode')"
                    class="text-left"
                  >{{$t('ClientID.tblPassCode')}}</th>
                  <th
                    :scope="$t('ClientID.tblGivenName')"
                    class="text-left"
                  >{{$t('ClientID.tblGivenName')}}</th>
                  <th
                    :scope="$t('ClientID.tblSurName')"
                    class="text-left"
                  >{{$t('ClientID.tblSurName')}}</th>
                </tr>
              </thead>
              <tbody>
                <tr v-for="(account) in getApiData.testAccount" :key="account.cardNumber">
                  <td>{{ account.cardNumber }}</td>
                  <td>{{ account.passcode }}</td>
                  <td>{{ account.attributes.givenname }}</td>
                  <td>{{ account.attributes.surname }}</td>
                </tr>
              </tbody>
            </template>
          </v-simple-table>
        </v-list-item-content>
      </v-list-item>
    </v-list>
  </v-card>
</template>
<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import Loading from '@/Atomic/Loading/Loading.vue';
const ClientIdModule = namespace('ClientIdModule');

@Component({
  components: {
    Loading
  }
})
export default class ClientIDDetails extends Vue {
  @Prop({ default: 0 })
  public id!: number;
  @ClientIdModule.Getter('errorStatus') public errorStatus!: boolean;
  @ClientIdModule.Action('getClientIdDetails')
  public getClientIdDetails!: any;
  @ClientIdModule.Getter('getApiData') public getApiData!: [];
  @ClientIdModule.Getter('isLoading') public isLoading!: boolean;

  private projectId: number = this.id || 0;
  private snackbar: boolean = false;
  private text: string = 'Copied';
  private timeout: number = 2000;

  public docopy(message: any) {
    this.$copyText(message);
    this.snackbar = true;
  }

  private mounted() {
    this.getClientIdDetails(this.projectId);
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
.active-bg {
  background-color: $BCgovActiveBg;
}
.pad-50 {
  padding-right: 50px;
}
</style>
