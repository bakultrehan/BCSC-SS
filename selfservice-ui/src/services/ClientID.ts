import axios from '@/lib/axios';
import { OIDCCONFIG_URL } from '@/config/api-endpoints';
import { getUrl } from '@/lib/helpers';

export default class ClientID {
  public static async getApiData(id: number) {
    const oidcURL = getUrl(OIDCCONFIG_URL, id);
    return await axios.get(oidcURL);
  }
}
