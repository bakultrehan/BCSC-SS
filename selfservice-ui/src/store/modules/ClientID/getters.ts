import { GetterTree } from 'vuex';
import { ClientIdState } from './types';
import { RootState } from '../../types';

/**
 * Getters Package
 */
export const getters: GetterTree<ClientIdState, RootState> = {
  /**
   * getApiData
   * @param {*} state
   * @returns  {object} package
   */

  getApiData(state) {
    return state.apiData;
  },
  successStatus(state) {
    return state.successStatus;
  },
  errorStatus(state) {
    return state.errorStatus;
  }
};
