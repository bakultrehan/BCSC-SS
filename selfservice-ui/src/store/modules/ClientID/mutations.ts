import { MutationTree } from 'vuex';
import { ClientIdState } from './types';

/**
 * mutation
 */
export const mutations: MutationTree<ClientIdState> = {
  /**
   * Sets loading
   * @param {*} state
   * @param {boolean} flag - loading
   */

  SET_LOADING(state, flag: boolean) {
    state.loading = flag;
  },
  /**
   * Sets package
   * @param {*} state
   * @param payload package list
   */

  SET_APIDETAILS(state, payload: any) {
    state.apiData = payload;
  }
};
