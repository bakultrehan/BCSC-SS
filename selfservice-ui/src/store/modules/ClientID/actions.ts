import { ActionTree } from 'vuex';
import { ClientIdState } from './types';
import { RootState } from '../../types';
// import i18n from '../../../i18n';

import ClientID from '@/services/ClientID';

/**
 * package Actions
 *
 */
export const actions: ActionTree<ClientIdState, RootState> = {
  /**
   * load getApiDetails from server and set to store
   * @param {*} { commit }
   */
  async getClientIdDetails({ commit }, id: number) {
    commit('SET_LOADING', true);
    const apiData = await ClientID.getApiData(id);

    commit('SET_APIDETAILS', apiData.data);
    commit('SET_LOADING', false);
  }
};
