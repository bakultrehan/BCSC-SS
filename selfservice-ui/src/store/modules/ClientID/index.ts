import { Module } from 'vuex';
import { getters } from './getters';
import { actions } from './actions';
import { mutations } from './mutations';
import { ClientIdState } from './types';
import { RootState } from '../../types';

/**
 * ClientIdModule state
 */
export const state: ClientIdState = {
  apiData: [],
  loading: true,
  successStatus: false,
  errorStatus: false,
  message: ''
};

const namespaced: boolean = true;

const ClientIdModule: Module<ClientIdState, RootState> = {
  namespaced,
  state,
  getters,
  actions,
  mutations
};

export default ClientIdModule;
