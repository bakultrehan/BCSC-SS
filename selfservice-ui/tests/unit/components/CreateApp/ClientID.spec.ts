import { mount } from '@vue/test-utils';
import ClientID from '@/components/CreateApp/ClientID.vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
let vuetify: any;
vuetify = new Vuetify();

describe('ClientID.vue', () => {
  const store = new Vuex.Store({
    modules: {
      ClientIdModule: {
        namespaced: true,
        state: {},
        getters: {
          isLoading: jest.fn(),
          getApiData: jest.fn(() => {
            return {
              clientId: 'abcd',
              clientSecret: 'test',
              testUserAccounts: [
                {
                  userName: 'test user',
                  idKey: 'test id key'
                },
                {
                  userName: 'test user',
                  idKey: 'test id key'
                }
              ]
            };
          }),
          errorStatus: jest.fn()
        },
        actions: {
          getClientIdDetails: jest.fn()
        }
      }
    }
  });

  const mountFunction = (options: any) => {
    return mount(ClientID, {
      store,
      vuetify,
      mocks: { $t: jest.fn(() => {}), $copyText: jest.fn(() => {}) }, // tslint:disable-line
      ...options
    });
  };

  it('renders props when passed with gettors', () => {
    const ClientDetails = mountFunction({});
    expect(ClientDetails.element).toMatchSnapshot();
  });

  it('renders props when passed with gettors', () => {
    const ClientDetails = mountFunction({});
    ClientDetails.find('.client-id-copy').trigger('click');
    expect(ClientDetails.element).toMatchSnapshot();
  });
});
