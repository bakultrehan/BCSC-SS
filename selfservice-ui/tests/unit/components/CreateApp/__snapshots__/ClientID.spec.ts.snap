// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientID.vue renders props when passed with gettors 1`] = `
<div
  class="mx-auto mb-5 v-card v-sheet theme--light"
>
  <!---->
   
  <!---->
   
  <header
    class="bc-subtitle-gold-1 v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--dense"
    style="height: 48px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 48px;"
    >
      <div
        class="v-card__title"
      >
        
      </div>
    </div>
  </header>
   
  <div
    class="v-list px-5 v-sheet v-sheet--tile theme--light v-list--dense"
    role="list"
  >
    <div
      class="v-list-item theme--light"
      role="listitem"
      tabindex="-1"
    >
      <div
        class="v-list-item__content"
      >
        
        
        
        <span
          class="small-hint pad-50"
        />
      </div>
       
      <div
        class="v-list-item__content align-end pointer font-weight-bold client-id-copy"
      >
        <div>
          <i
            aria-hidden="true"
            class="v-icon notranslate mr-1 mdi mdi-content-copy theme--light"
            style="font-size: 16px;"
          />
          
          abcd
        
        </div>
      </div>
    </div>
     
    <hr
      aria-orientation="horizontal"
      class="v-divider theme--light"
      role="separator"
    />
     
    <div
      class="v-list-item theme--light"
      role="listitem"
      tabindex="-1"
    >
      <div
        class="v-list-item__content"
      >
        
        
        
        <span
          class="small-hint pad-50"
        />
      </div>
       
      <div
        class="v-list-item__content align-end pointer font-weight-bold d-block"
      >
        <div
          class="float-left"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mr-2 mdi mdi-content-copy theme--light"
            style="font-size: 16px;"
          />
        </div>
         
        <div>
          <span>
            test
          </span>
        </div>
      </div>
    </div>
     
    <hr
      aria-orientation="horizontal"
      class="v-divider theme--light"
      role="separator"
    />
     
    <div
      class="v-list-item theme--light"
      role="listitem"
      tabindex="-1"
    >
      <div
        class="v-list-item__content align-self-start"
      >
        
        
        
        <span
          class="small-hint pad-50"
        />
      </div>
       
      <div
        class="v-list-item__content align-self-start"
      >
        <div
          class="v-data-table v-data-table--dense theme--light"
        >
          <div
            class="v-data-table__wrapper"
          >
            <table>
              <thead>
                <tr>
                  <th
                    class="text-left"
                  >
                    
                  </th>
                   
                  <th
                    class="text-left"
                  >
                    
                  </th>
                   
                  <th
                    class="text-left"
                  >
                    
                  </th>
                   
                  <th
                    class="text-left"
                  >
                    
                  </th>
                </tr>
              </thead>
               
              <tbody />
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClientID.vue renders props when passed with gettors 2`] = `
<div
  class="mx-auto mb-5 v-card v-sheet theme--light"
>
  <div
    class="v-snack v-snack--active v-snack--bottom"
    name="v-snack-transition"
  >
    <div
      class="v-snack__wrapper"
      role="alert"
    >
      <div
        class="v-snack__content"
      >
        
    Copied
    
        <button
          class="v-btn v-btn--flat v-btn--text theme--light v-size--default blue--text"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            
          </span>
        </button>
      </div>
    </div>
  </div>
   
  <!---->
   
  <header
    class="bc-subtitle-gold-1 v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--dense"
    style="height: 48px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 48px;"
    >
      <div
        class="v-card__title"
      >
        
      </div>
    </div>
  </header>
   
  <div
    class="v-list px-5 v-sheet v-sheet--tile theme--light v-list--dense"
    role="list"
  >
    <div
      class="v-list-item theme--light"
      role="listitem"
      tabindex="-1"
    >
      <div
        class="v-list-item__content"
      >
        
        
        
        <span
          class="small-hint pad-50"
        />
      </div>
       
      <div
        class="v-list-item__content align-end pointer font-weight-bold client-id-copy"
      >
        <div>
          <i
            aria-hidden="true"
            class="v-icon notranslate mr-1 mdi mdi-content-copy theme--light"
            style="font-size: 16px;"
          />
          
          abcd
        
        </div>
      </div>
    </div>
     
    <hr
      aria-orientation="horizontal"
      class="v-divider theme--light"
      role="separator"
    />
     
    <div
      class="v-list-item theme--light"
      role="listitem"
      tabindex="-1"
    >
      <div
        class="v-list-item__content"
      >
        
        
        
        <span
          class="small-hint pad-50"
        />
      </div>
       
      <div
        class="v-list-item__content align-end pointer font-weight-bold d-block"
      >
        <div
          class="float-left"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mr-2 mdi mdi-content-copy theme--light"
            style="font-size: 16px;"
          />
        </div>
         
        <div>
          <span>
            test
          </span>
        </div>
      </div>
    </div>
     
    <hr
      aria-orientation="horizontal"
      class="v-divider theme--light"
      role="separator"
    />
     
    <div
      class="v-list-item theme--light"
      role="listitem"
      tabindex="-1"
    >
      <div
        class="v-list-item__content align-self-start"
      >
        
        
        
        <span
          class="small-hint pad-50"
        />
      </div>
       
      <div
        class="v-list-item__content align-self-start"
      >
        <div
          class="v-data-table v-data-table--dense theme--light"
        >
          <div
            class="v-data-table__wrapper"
          >
            <table>
              <thead>
                <tr>
                  <th
                    class="text-left"
                  >
                    
                  </th>
                   
                  <th
                    class="text-left"
                  >
                    
                  </th>
                   
                  <th
                    class="text-left"
                  >
                    
                  </th>
                   
                  <th
                    class="text-left"
                  >
                    
                  </th>
                </tr>
              </thead>
               
              <tbody />
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
